src/server/rpc/procedures/weather/weather.js

```diff
@@ -23,189 +21,113 @@ var isWithinMaxDistance = function(result, lat, lng) {
                 humidity = body.main.humidity;
             }
-            return response.json(humidity);
+            this.response.send(humidity);
         });
+};
 
-        return null;
-    },
-
-    description: function(latitude, longitude) {
-        var url = baseUrl + '&lat=' + latitude + '&lon=' + longitude,
-            response = this.response;
 
-        request(url, (err, res, body) => {
-            if (err || res.statusCode < 200 || res.statusCode > 299) {
-                log('ERROR: ', (err || body));
-                return response.status(500).send('ERROR: '+(err || body));
-            }
-            body = JSON.parse(body);
+weather.description = function(latitude, longitude){
+    return this._requestData({queryString: '&lat=' + latitude + '&lon=' + longitude})
+        .then(body => {
             var description = 'unknown';
             if (isWithinMaxDistance(body, latitude, longitude)) {
                 description = body.weather[0].description;
             }
-            return response.json(description);
+            this.response.send(description);
         });
+};
 
-        return null;
-    },
-
-    icon: function(latitude, longitude) {
-        var url = baseUrl + '&lat=' + latitude + '&lon=' + longitude,
-            response = this.response;
+weather.windSpeed = function(latitude, longitude){
+    return this._requestData({queryString: '&lat=' + latitude + '&lon=' + longitude})
+        .then(body => {
+            var speed = 'unknown';
+            if (isWithinMaxDistance(body, latitude, longitude)) {
+                speed = body.wind.speed || 'unknown';
+            }
+            this.response.send(speed);
+        });
+};
 
-        request(url, (err, res, body) => {
-            if (err || res.statusCode < 200 || res.statusCode > 299) {
-                log('ERROR: ', (err || body));
-                return response.status(500).send('ERROR: '+(err || body));
+weather.windAngle = function(latitude, longitude){
+    return this._requestData({queryString: '&lat=' + latitude + '&lon=' + longitude})
+        .then(body => {
+            var deg = 'unknown';
+            if (isWithinMaxDistance(body, latitude, longitude)) {
+                deg = body.wind.deg || 'unknown';
             }
-            body = JSON.parse(body);
+            this.response.send(deg);
+        });
+};
+
+weather.icon = function(latitude, longitude){
+    return this._requestData({queryString: '&lat=' + latitude + '&lon=' + longitude})
+        .then(body => {
             // Return sunny if unknown
             var iconName = '01d.png';
             if (body.weather && body.weather[0]) {
                 iconName = body.weather[0].icon+'.png';
             }
-            request.get(baseIconUrl+iconName).pipe(response);
+            let queryOpts = {
+                queryString: iconName,
+                baseUrl: 'http://openweathermap.org/img/w/',
+                cache: false
```

Comment:
I remember why I didn't cache it, the TTL for the cache store on this service is set to 60sec and it's kinda low and since the call to get the image has no rate limit I didn't include it in the cache. but it still could be useful